### ISSUE TITLE
Incorrect content ids on hot reload

### ISSUE THREAD

**Issue by jonathan-lws** [nuxt/content#2872]
### Environment

- Operating System: Windows_NT
- Node Version:     v20.17.0
- Nuxt Version:     3.14.1592
- CLI Version:      3.15.0
- Nitro Version:    2.10.4
- Package Manager:  npm@10.8.2
- Builder:          -
- User Config:      default
- Runtime Modules:  @nuxtjs/i18n@9.1.0, @nuxt/eslint@0.7.1, @nuxt/ui@3.0.0-alpha.9, @nuxt/content@3.0.0-alpha.7, @nuxt/image@1.8.1, @nuxt/scripts@0.9.5, @nuxtjs/algolia@1.10.2
- Build Modules:    -

### Version

v3.0.0-alpha.7

### Reproduction

- Create a content collection `homepage` (should be the same with any naming)

```
// ---
homepage: defineCollection({
    type: 'page',
    source: '**/1.homepage/*.md',
  })
// ---
```

Write some contents in the directory then render it with ContentRenderer.
When running the app, the sqllite database is successfully created, the content is correctly rendering.

Hot reload is not working.

### Description

The problem is that hot reload is not working.

After investigations, i decided to open the sqlite database with HeidiSQL to watch the changes in real time when i do changes in .md files.

I notices after a hot reload, the line concerning the content that was updated is duplicated with a different id.
I think it's due to the fact that i'm on Windows.

The correct id like generated by the Nuxt build process is `homepage/fr/1.homepage/10.headline.md`.
However, after a hot reload, the duplicated line have ID `homepage\fr\1.homepage\10.headline.md`.

This is my first issue reporting.
Please excuse me in advance if i made mistakes and please tell me if so.

Thanks for your wonderfull job

### Additional context

I guess it occurs exclusively on Windows

I think paths should be normalized from win32 to posix when the are related to the database in order to avoid differences.
We can use https://github.com/unjs/pathe/blob/main/src/path.ts#L23 for normalizing paths (inside code watchers)

**Comment by farnabaz** [comment]
It should be fixed in f883273, Could you confirm the fix using `npm i https://pkg.pr.new/@nuxt/content@f883273`

**Comment by jonathan-lws** [comment]
Thanks, i confirm the problem is solved 👍

**Comment by farnabaz** [comment]
Released in [v3.0.0-alpha.8](https://github.com/nuxt/content/releases/tag/v3.0.0-alpha.8).
